turbinia/state_manager.py

```diff
@@ -328,3 +332,146 @@ def write_new_task(self, task):
+
+    Args:
+      evidence_dict (dict[str]): A dictionary containing the serialized
+      evidence attributes that will be saved.
+
+    Returns:
+      key (str): The key corresponding to the evidence in redis
+    """
+    key = ':'.join(('TurbiniaEvidence', json.loads(evidence_dict['id'])))
+    log.info(f'Writing new evidence {key} into Redis')
+    for attribute_key, attribute_value in evidence_dict.items():
+      if not self.client.hset(key, attribute_key, attribute_value):
+        log.error(
+            f'Unsuccessful in writing {attribute_key} of {key} into Redis')
+    if evidence_hash := json.loads(evidence_dict.get('hash')):
+      self.client.hset('TurbiniaEvidenceHashes', evidence_hash, key)
+    return key
+
+  def update_evidence_attribute(
+      self, evidence_id: str, attribute_name: str, json_value: str):
+    """Updates one attribute of the evidence in Redis.
+
+    Args:
+      evidence_id (str): The ID of the stored evidence.
+      attribute_name (str): name of the attribute to be updated.
+      json_value (str): json value to be updated.
+    """
+    key = ':'.join(('TurbiniaEvidence', evidence_id))
+    log.info(f'Updating attribute {attribute_name} for evidence {key} in Redis')
+    if self.client.hset(key, attribute_name, json_value):
+      if attribute_name == 'hash' and json_value:
+        self.client.hset('TurbiniaEvidenceHashes', json.loads(json_value), key)
+
+  def get_evidence(self, evidence_id: str):
+    """Gets one evidence from Redis given its ID.
+
+    Args:
+      evidence_id (str): The ID of the stored evidence.
+
+    Returns:
+      evidence_dict (dict): Dict containing evidence attributes. 
+    """
+    key = ':'.join(('TurbiniaEvidence', evidence_id))
+    evidence_keys = self.client.hkeys(key)
+    evidence_dict = {}
+    for attribute_key in evidence_keys:
+      evidence_dict[attribute_key.decode()] = json.loads(
+          self.client.hget(key, attribute_key))
+    return evidence_dict
+
+  def evidence_exists(self, evidence_id):
+    """Checks if the evidence is saved in Redis given its ID.
+
+    Args:
+      evidence_id (str): The ID of the stored evidence.
+
+    Returns:
+      evidence_id (bool): Boolean indicating if evidence is saved. 
+    """
+    return self.client.exists(':'.join(('TurbiniaEvidence', evidence_id)))
+
+  def get_evidence_summary(self, sort: str = None, output: str = 'keys'):
+    """Gets a summary of all evidences.
+
+    Args:
+      sort (str): Name of the evidence attribute used to sort the output.
+      output (str): Value to be output by the function (keys | values | count).
+
+    Returns:
+      summary (dict | int): Object containing evidences. 
+    """
+    if output == 'count' and not sort:
+      return sum(1 for _ in self.client.scan_iter('TurbiniaEvidence:*'))
+    summary = {} if sort else []
+    for key in self.client.scan_iter('TurbiniaEvidence:*'):
+      value = self.get_evidence(key.decode().split(':')[1])
```

Comment:
The variable name ```value``` is a bit ambigious here, could we name this something like ```evidence_dictionary```?

Reply:
Thanks, I renamed now